src/DotNetProjectFile.Analyzers/CodeAnalysis/Language.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Frozen;
+
+namespace DotNetProjectFile.CodeAnalysis;
+
+/// <summary>
+/// Represents a language that is supported by the .NET project file analyzers.
+/// </summary>
+public readonly struct Language : IEquatable<Language>
+{
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    private readonly string Id;
+
+    private Language(string id) => Id = id;
+
+    /// <summary>None/unknown langauge.</summary>
+    public static readonly Language None;
+
+    /// <summary>C#.</summary>
+    public static readonly Language CSharp = new(nameof(CSharp));
+
+    /// <summary>Visual Basic (.NET).</summary>
+    public static readonly Language VisualBasic = new(nameof(VisualBasic));
+
+    /// <summary>F#.</summary>
+    public static readonly Language FSharp = new(nameof(FSharp));
+
+    /// <summary>Gets the MSBuild project file extenion for the language.</summary>
+    public string? ProjectFile => Id switch
```

Comment:
Would it maybe be better to have these fields/props be configured through the constructor instead?

Reply:
Done.